client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/ClientMetricsReporter.java

```diff
@@ -98,17 +105,26 @@ public void validateReconfiguration(Map<String, ?> configs) throws ConfigExcepti
 
     @Override
     public Set<String> reconfigurableConfigs() {
-        return Collections.emptySet();
+        return Set.of();
     }
 
     @Override
     public void contextChange(MetricsContext metricsContext) {
         String prefix = metricsContext.contextLabels().get(MetricsContext.NAMESPACE);
+        if (!PREFIXES.contains(prefix)) {
+            throw new IllegalStateException("ClientMetricsReporter should only be used in Kafka servers");
```

Comment:
ClientMetricsReporter should only be used by clients?

Reply:
Yep, fixed.